Core/include/Acts/Propagator/MultiEigenStepperLoop.ipp

```diff
@@ -0,0 +1,200 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+namespace Acts {
+
+template <typename E, typename R, typename A>
+auto MultiEigenStepperLoop<E, R, A>::boundState(State& state,
+                                                const Surface& surface,
+                                                bool transportCov) const
+    -> Result<BoundState> {
+  if (numberComponents(state) == 1) {
+    return SingleStepper::boundState(state.components.front().state, surface,
+                                     transportCov);
+  } else {  // Do the combination
+    SmallVector<std::pair<double, BoundTrackParameters>> states;
+    double accumulatedPathLength = 0.0;
+    int failedBoundTransforms = 0;
+
+    for (auto i = 0ul; i < numberComponents(state); ++i) {
+      auto bs = SingleStepper::boundState(state.components[i].state, surface,
+                                          transportCov);
+
+      if (bs.ok()) {
+        states.push_back(
+            {state.components[i].weight, std::get<BoundTrackParameters>(*bs)});
+        accumulatedPathLength +=
+            std::get<double>(*bs) * state.components[i].weight;
+      } else {
+        failedBoundTransforms++;
+      }
+    }
+
+    if (failedBoundTransforms > 0) {
+      ACTS_ERROR("Multi component bound state: "
+                 << failedBoundTransforms << " of " << numberComponents(state)
+                 << " transforms failed");
+    }
+
+    if (states.size() == 0) {
+      return MultiStepperError::AllComponentsConversionToBoundFailed;
+    }
+
+    // TODO also implement a method of using the mode of the mixture
```

Comment:
I don't understand the comment here - do you want to make this configurable?

Reply:
That was one thing that came up in the meeting with Anthony and Christos, they said that at ATLAS they use the mode of the mixture for the final parameters, not the mean. Currently the bound states are combined by computing the mean. This probably should be tried out during the validation phase?

However, I will clarify this comment with a better formulation